db.go

```diff
@@ -934,6 +934,26 @@ func (db *DB) Begin() (*Tx, error) {
 	return &Tx{Tx: tx, DB: db}, nil
 }
 
+// Transaction begins a transaction as a block, if it errors it
+// will rollback and return an error, otherwise it will commit.
+func (db *DB) Transaction(fn func(tx *Tx) error) error {
+	tx, err := db.Begin()
+
+	if err != nil {
+		return err
+	}
+
+	if err := fn(tx); err != nil {
```

Comment:
Would like to see panic recovery since we are calling an externally-supplied function.
Rollback can be done in a `defer` clause - if `tx.Commit()` happens first then `tx.Rollback` will be a no-op.

Reply:
Sounds good. I'll make the changes.

Reply:
I handled the rollback as well as the recovery on the defer clause. Howver, through my testing if the transaction has been committed `tx.Rollback()` will error out so that needed to be handled as well